apps/backend/src/reconciliation/pos-reconciliation.service.ts

```diff
@@ -0,0 +1,60 @@
+import { Injectable, Logger, Inject } from '@nestjs/common';
+import { PaymentEntity } from '../sales/entities/payment.entity';
+import { POSDepositEntity } from './../deposits/entities/pos-deposit.entity';
+import { SalesService } from '../sales/sales.service';
+import { AppLogger } from '../common/logger.service';
+import {
+  ReconciliationEvent,
+  ReconciliationEventOutput,
+  ReconciliationEventError
+} from './const';
+import { PosDepositService } from '../deposits/pos-deposit.service';
+@Injectable()
+export class POSReconciliationService {
+  constructor(
+    @Inject(Logger) private readonly appLogger: AppLogger,
+    @Inject(PosDepositService) private posDepositService: PosDepositService,
+    @Inject(SalesService) private salesService: SalesService
+  ) {}
+
+  private async match(
+    deposits: POSDepositEntity[],
+    payments: PaymentEntity[]
+  ): Promise<any[]> {
+    return payments.map(async (payment: PaymentEntity) => {
+      return await this.setMatched(
+        deposits.find(
+          (deposit: POSDepositEntity) =>
+            parseFloat(deposit.transaction_amt.toString()) ===
```

Comment:
this looks strange, add a todo on what types need to be updated on the entity and the dtos